### PR TITLE
[migration phase 1] PodFitsHostPorts as filter plugin

### LINKED ISSUE #83632: [migration phase 1] PodFitsHostPorts as filter plugin
**What would you like to be added**:

Add a filter plugin that calls into PodFitsHostPorts predicate, example #83460

**Why is this needed**:

Part of  #83554

/sig scheduling
/help
/priority important-soon

### PR DESCRIPTION
**Pull request by wgliang**
**What type of PR is this?**

/kind cleanup

/sig scheduling
/priority important-soon

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #83632

**Special notes for your reviewer**:
/assign @ahg-g 

**Does this PR introduce a user-facing change?**:

```release-note
[migration phase 1] PodFitsHostPorts as filter plugin
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

**Review by ahg-g — COMMENTED** [review]
ditto to copying unit tests.

**Review by ahg-g — COMMENTED** [review]
Thanks Leon, ditto to copying unit tests.

**Comment by wgliang** [comment]
@ahg-g Ready for review. Thanks.

File: pkg/scheduler/framework/plugins/podfitshostports/pod_fits_host_ports_test.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+			name:       "UDP hostPort conflict due to 0.0.0.0 hostIP",
+			wantStatus: framework.NewStatus(framework.Unschedulable, predicates.ErrPodNotFitsHostPorts.GetReason()),
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			meta := predicates.GetPredicateMetadata(test.pod, nil)
+			state := framework.NewCycleState()
+			state.Write(migration.PredicatesStateKey, &migration.PredicatesStateData{Reference: meta})
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/podfitshostports/pod_fits_host_ports_test.go]: the predicate doesn't read metadata or state, you could remove all three lines above and pass nil to Filter as state.

**Review by ahg-g — COMMENTED** [review]
it seems redundant that we prefix the filter name with "podfits", everything we do is for the pod, lets name this plugin ```NodePorts```

/cc @alculquicondor @Huang-Wei

**Comment by wgliang** [comment]
Ok, these are not synced here. 
Are there other comments? I can solve it together, thanks.

**Comment by wgliang** [comment]
@ahg-g 
Ok, these are not synced here.
Are there other comments? I can solve it together, thanks.

**Comment by ahg-g** [comment]
> @ahg-g
> Ok, these are not synced here.
> Are there other comments? I can solve it together, thanks.

Nothing for now, it wasn't my intention to follow up with more comments, just happened that I noticed more things after the first round :)

File: pkg/scheduler/framework/plugins/podfitshostports/pod_fits_host_ports_test.go (hunk 1 above)
Comment (wgliang) [reply pkg/scheduler/framework/plugins/podfitshostports/pod_fits_host_ports_test.go]: done

**Comment by wgliang** [comment]
Cool, Ready for review. Please take a look, thanks. @ahg-g

**Pull request merged**